project-base/storefront/pages/order/contact-information.tsx

```diff
@@ -88,16 +89,14 @@ const ContactInformationPage: FC<ServerSidePropsType> = () => {
     }, []);
 
     useEffect(() => {
-        if (!currentCart.cart?.items.length) {
-            router.replace(cartUrl);
-        }
-    }, [currentCart.cart]);
-
-    useEffect(() => {
-        if (!currentCart.transport || !currentCart.payment) {
-            router.replace(transportAndPaymentUrl);
+        if (currentCart.cart !== undefined) {
+            if (!currentCart.cart?.items.length) {
+                router.replace(cartUrl);
+            } else if (!(currentCart.transport && currentCart.payment)) {
+                router.replace(transportAndPaymentUrl);
+            }
         }
```

Comment:
I do not like nested conditions like this. I am not sure what your source for early returns from useEffects has been, but based on the type definition for useEffects, I do not see anything wrong with using an early return given that there is no destructor provided. If I base my comment on the fact that I have not read anything about it being wrong, I would be glad if you could unwrap the conditions.

Reply:
We will discuss this on FE meeting. For now, skipping this.